File: benchmark_release/public/case_000027/src/app/services/auditService.js
'use strict';

const { db } = require('../db');

function log(actorId, action, detail = '') {
  try {
    db.prepare('INSERT INTO audit_log (actor_id, action, detail) VALUES (?,?,?)')
      .run(actorId, action, String(detail).slice(0, 500));
  } catch (e) {
    console.error('[audit] write failed:', e.message);
  }
}

function getRecent(limit = 100) {
  return db.prepare(
    `SELECT l.id, l.action, l.detail, l.created_at, u.username
     FROM audit_log l LEFT JOIN users u ON l.actor_id = u.id
     ORDER BY l.created_at DESC LIMIT ?`
  ).all(limit);
}

module.exports = { log, getRecent };

File: benchmark_controls_release/public/case_000078_clean/src/services/audit_service.py
"""
AuditService — thin wrapper around audit_log writes.
TODO: add severity levels (INFO / WARN / CRITICAL) for SIEM integration.
"""
import sqlite3

DB_PATH = "/tmp/filesync.db"


def log_event(user_id: int, action: str, detail: str, ip: str, severity: str = "INFO") -> None:
    """
    Persists an audit event.  severity is reserved for future SIEM routing
    and not yet stored in the schema (schema migration pending).
    """
    try:
        conn = sqlite3.connect(DB_PATH)
        c = conn.cursor()
        c.execute(
            "INSERT INTO audit_log (user_id, action, detail, ip_address) VALUES (?, ?, ?, ?)",
            (user_id, action, detail, ip)
        )
        conn.commit()
        conn.close()
    except Exception:
        # NOTE: swallowing here so audit failures never break the request path
        pass


def recent_events(limit: int = 50) -> list:
    conn = sqlite3.connect(DB_PATH)
    c = conn.cursor()
    c.execute("""SELECT a.id, u.username, a.action, a.detail, a.ip_address, a.created_at
                 FROM audit_log a LEFT JOIN users u ON a.user_id=u.id
                 ORDER BY a.id DESC LIMIT ?""", (limit,))
    rows = c.fetchall()
    conn.close()
    return [{"id": r[0], "username": r[1], "action": r[2],
             "detail": r[3], "ip": r[4], "timestamp": r[5]} for r in rows]